libs/newsletters-data-client/src/lib/newsletter-storage/InMemoryNewsletterStorage.ts

```diff
@@ -154,6 +154,28 @@ export class InMemoryNewsletterStorage implements NewsletterStorage {
 		return Promise.resolve(response);
 	}
 
+	async replace(
+		listId: number,
+		newsletter: NewsletterDataWithoutMeta,
+		user: UserProfile,
+	) {
+		const match = this.memory.find((item) => item.listId === listId);
+		if (!match) {
+			return Promise.resolve(this.buildNoItemError(listId));
+		}
+
```

Comment:
There is a case for throwing an error if `newsletter.identityName !== match.identityName || newsletter.listId !== match.listId`.